File: Python/programmers/etc/level2/큰_수_만들기.py
def solution(number, k):
    answer = []
    
    for num in number:
        while k > 0 and answer and answer[-1] < num:
            answer.pop()
            k -= 1
        answer.append(num)

    return "".join(answer[:len(answer)-k])


number, k = "1924", 2
print(solution(number, k)) # "94"

number, k = "1231234", 3
print(solution(number, k)) # "3234"

number, k = "4177252841", 4
print(solution(number, k)) # "775841"

File: Java/programmers/etc/level2/큰수만들기.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 큰수만들기 {
    public String solution(String number, int k) {
        Stack<Character> tmp = new Stack<>();

        for (char num : number.toCharArray()) {
            while (k>0 && !tmp.isEmpty() && tmp.peek() < num) {
                tmp.pop();
                k--;
            }
            tmp.push(num);
        }

        String answer = "";

        for (int i=0; i<tmp.size()-k; i++) {
            answer += tmp.get(i);
        }

        return answer;
    }
}
